lms/static/scripts/frontend_apps/components/Table.js

```diff
@@ -24,8 +29,34 @@ function nextItem(items, currentItem, step) {
   return items[index + step];
 }
 
+/**
+ * @typedef TableColumn
+ * @prop {string} label - Header label for the column
+ * @prop {string} className - Additional classes for the column's `<th>` element
+ */
+
+/**
+ * @template Item
+ * @typedef TableProps
+ * @prop {string} accessibleLabel - An accessible label for the table.
+ * @prop {TableColumn[]} columns - The columns to display in this table.
+ * @prop {Item[]} items - The items to display in this table.
+ * @prop {(it: Item, selected: boolean) => any} renderItem -
```

Comment:
I feel like naming this `renderItem` may be an extra level of abstraction (and required me to read code source to understand what it meant). Would `renderRow` be too specific?